add spec for punctuatedName and namedType #709 (#710)
graphql/graphql-spec: spec/Section 4 -- Introspection.md

* `kind` must return `__TypeKind.SCALAR`.
* `name` must return a String.
* `description` may return a String or {null}.

> * `punctuatedName` must return `name`.
> * `namedType` must return a reference to the __Type object {this}.

Reviewer: `{this}` doesn't have any specific meaning in the GraphQL spec - we must be careful not to use JavaScript idioms.

```suggestion
* `punctuatedName` must return the same value as the `name` field would.
* `namedType` must return this type.
```

* `kind` must return `__TypeKind.LIST`.
* `ofType`: Any type.

> * `punctuatedName` must return `'[' + ofType.punctuatedName + ']'`.

Reviewer: @IvanGoncharov Using `+` for string concatenation in the spec is likely not allowed since it's a cross-language specification. Do we have any examples in the spec where concatenation is used that we could apply here?

Reviewer: Same thing goes for `ofType.punctuatedName` for referencing a field of a value. I think we might have to settle for writing this in prose, awkward though this may be:

```suggestion
* `punctuatedName` must return the result of concatenating the `[` character, the wrapped type's `punctuatedName` value, and a `]` character.
```
